src/container/container.go

```diff
@@ -0,0 +1,272 @@
+	stats, _ := cli.ContainerStats(ctx, cid, false)
+	data := types.StatsJSON{}
+	err = json.NewDecoder(stats.Body).Decode(&data)
+	if err != nil {
+		return metrics, err
+	}
+
+	// Calculate CPU  and per CPU percent
+	cpuPercent := 0.0
+	numCPUs := len(data.CPUStats.CPUUsage.PercpuUsage)
+
+	cpuDelta := float64(data.CPUStats.CPUUsage.TotalUsage) - float64(data.PreCPUStats.CPUUsage.TotalUsage)
+
+	systemDelta := float64(data.CPUStats.SystemUsage) - float64(data.PreCPUStats.SystemUsage)
+
+	if cpuDelta > 0.0 && systemDelta > 0.0 {
+		cpuPercent = (cpuDelta / systemDelta) * float64(numCPUs) * 100.0
+	}
+
+	// Get Per CPU utilizations
+	preLen := len(data.PreCPUStats.CPUUsage.PercpuUsage)
+	postLen := len(data.CPUStats.CPUUsage.PercpuUsage)
+	numCPUs = ui.MaxInt(preLen, postLen)
+
+	perCpuPercents := make([]string, numCPUs)
+
+	// If first run, skip percpu metrics
+	if preLen != postLen {
+		for i := range perCpuPercents {
+			perCpuPercents[i] = "NA"
+		}
+	} else {
+		for i, usage := range data.CPUStats.CPUUsage.PercpuUsage {
+			perCpuPercent := 0.0
+
+			cpuDelta := float64(usage) - float64(data.PreCPUStats.CPUUsage.PercpuUsage[i])
+
+			if cpuDelta > 0.0 && systemDelta > 0.0 {
+				perCpuPercent = (cpuDelta / systemDelta) * float64(numCPUs) * 100.0
+			}
+			perCpuPercents[i] = fmt.Sprintf("%.2f%%", perCpuPercent)
+		}
+	}
+
+	// Calculate Memory usage
+	memPercent := float64(data.MemoryStats.Usage) / float64(data.MemoryStats.Limit) * 100
+
+	// Calculate Network Metrics
+	var rx, tx float64
+	for _, v := range data.Networks {
+		rx += float64(v.RxBytes)
+		tx += float64(v.TxBytes)
+	}
+
+	// Calculate blk IO
+	var blkRead, blkWrite uint64
+	for _, bioEntry := range data.BlkioStats.IoServiceBytesRecursive {
+		switch strings.ToLower(bioEntry.Op) {
+		case "read":
+			blkRead = blkRead + bioEntry.Value
+		case "write":
+			blkWrite = blkWrite + bioEntry.Value
+		}
+	}
+
+	// Get Network Settings
+	netData := []netInfo{}
+	for _, network := range c.NetworkSettings.Networks {
+		id := network.NetworkID
+
+		net, err := cli.NetworkInspect(ctx, id, types.NetworkInspectOptions{})
+		if err != nil {
+			continue
+		}
+
+		n := netInfo{
+			Name:    net.Name,
+			Driver:  net.Driver,
+			Ip:      network.IPAddress,
+			Ingress: net.Ingress,
+		}
+
+		netData = append(netData, n)
+	}
+
+	// Get Port mappings
+	portData := []portMap{}
+	for _, port := range c.Ports {
+		p := portMap{
+			Host:      int(port.PublicPort),
+			Container: int(port.PrivatePort),
+			Protocol:  port.Type,
+		}
+
+		portData = append(portData, p)
+	}
+
+	// Get mounted volumes
+	mountData := []mountInfo{}
+	for _, mount := range c.Mounts {
+		m := mountInfo{
+			Src:  mount.Source,
+			Dst:  mount.Destination,
+			Mode: mount.Mode,
+		}
+
+		mountData = append(mountData, m)
+	}
+
+	// Get processes in container
+	procs, err := cli.ContainerTop(ctx, cid, []string{})
+	if err != nil {
+		return metrics, nil
+	}
+
+	procData := []procInfo{}
+	for _, proc := range procs.Processes {
```

Comment:
`proc[0]`, etc isn't very understandable. My suggestion would be to have different functions called `get<whatever proc[0] is supposed to give>()` or something similar. The reason I suggest seperate functions is mainly because the implementation might change in the future, in case it does, incorporating that change won't be too much of a hastle. 

But, if you think that having it the way it is is better, then it might be a good idea to maybe provide comments on what those fields exactly are.

Reply:
The docker engine API is defined this way, I see no reason why we'd want to create functions just to extract these values. As for the fields, they're named in the struct itself, so that should be clear in itself right?